### ISSUE TITLE
Interactive search in artist list doesn't work anymore

### ISSUE THREAD

**Issue by rien333** [SoongNoonien/mpdevil#86]
New — and so far very happy — user here! ☺️

The [usage docs](https://github.com/SoongNoonien/mpdevil/wiki/Usage) state that:

> **Artist List**
> Just start typing to open an interactive search. Artist names that begin with the search string will be selected.

This indeed seems to be the default behavior in the latest stable release (1.11.0). However, I'm currently running from the latest commit, and this behavior now seems to be entirely absent.

Is this intentional? To be honest, I have slightly too many artists to use `mpdevil` without this feature. Scrolling, on its own, is just is not precise or fast enough to find something specific, and the search view 1) requires more clicks 2) returns impractically long list of results for certain queries.

(I believe that in previous versions, the interactive search was triggered by focusing the artist list widget, and then starting to type. If this  exact behavior is no longer possible due to  recent rewrites, perhaps one could add a keyboard shortcut to emulate this now-removed behavior?)

**Comment by SoongNoonien** [comment]
Hi!
The docs are a bit out of date, I need to fix them at some point in time. But the search feature should be readded when I get to it. Gtk4 does some things quite differently including lists and the interactive search in lists so the feature got lost while porting. Ideally the old interactive search should be replaced by a filter system which only shows artists matching the input string.

> If this exact behavior is no longer possible due to recent rewrites, perhaps one could add a keyboard shortcut to emulate this now-removed behavior?

The usual shortcut used for this would be `<ctrl>f` but this is currently used by the title/song search so this needs a redesign first.

**Comment by rien333** [comment]
Sounds good! Leaving this open as a bit of a warning to others, but feel free to close.

**Comment by SoongNoonien** [comment]
What do you think about the new search? You can now search via the usual `<ctrl>f`, the search button or by simply typing. When you click on a song you will be redirected to this song in the normal browser. This way it should also be relatively easy to find artists or albums.
I decided against displaying artists and albums as search results too, since it is currently not possible to perform a case-insensitive search for artists or albums directly with mpd (see: https://github.com/MusicPlayerDaemon/MPD/issues/494).

**Comment by rien333** [comment]
Yeah, the new search is pretty neat! More consistent in terms of UX/UI, especially. I like the newly added fuzzy search as well — searching for "[artist name] [album name]" is quite efficient.  

That said, I  still might prefer the search to return albums/artists instead of songs, since I 1) mostly listen to full albums anyway, and 2) the search result get crowded pretty quickly.  Then again, maybe I will get used to how search works now! (and I understand that this  is currently a limit with mpd)

 Thanks for all your work as of late :)  I hope to contribute something someday, if only because I feel like you made a good base.

**Comment by SoongNoonien** [comment]
As suggested in https://github.com/SoongNoonien/mpdevil/issues/88#issuecomment-1975394899 I've added an artist search to the search window. What do you think about it? I think the keyboard navigation could still be improved, I'd like to be able to use the arrow keys to switch between the artists and the songs. This shouldn't be that hard to implement though.

@Richjn I've tried to test the feasibility for large collections and I think it should be fast enough. I don't actually have that large of a collection so I had to make a "simulation". How does it perform with a genuine large collection like yours?

**Comment by Richjn** [comment]
Performance is very good!
I have two PC's my main box is quite high end and there is no discernible lag at all in the search, my secondary box is a quite low spec Intel NUC and performance there was also very good with virtually no lag.

Thanks for adding this feature it makes a huge difference in usability for those with large collections :-)

**Comment by SoongNoonien** [comment]
> Performance is very good!

Ok, nice!
